lib/auth/touchid/authenticate.m

```diff
@@ -0,0 +1,61 @@
+// go:build touchid
+//  +build touchid
+
+// Copyright 2022 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "authenticate.h"
+
+#import <CoreFoundation/CoreFoundation.h>
+#import <Foundation/Foundation.h>
+#import <Security/Security.h>
+
+#include "common.h"
+
+int Authenticate(AuthenticateRequest req, char **sigB64Out, char **errOut) {
+  NSData *appLabel = [NSData dataWithBytes:req.app_label
+                                    length:strlen(req.app_label)];
+  NSDictionary *query = @{
+    (id)kSecClass : (id)kSecClassKey,
+    (id)kSecAttrKeyType : (id)kSecAttrKeyTypeECSECPrimeRandom,
+    (id)kSecMatchLimit : (id)kSecMatchLimitOne,
+    (id)kSecReturnRef : @YES,
+    (id)kSecAttrApplicationLabel : appLabel,
+  };
+  SecKeyRef privateKey = NULL;
+  OSStatus status = SecItemCopyMatching((__bridge CFDictionaryRef)query,
+                                        (CFTypeRef *)&privateKey);
```

Comment:
Do we need to add `bridge_retained` to this cast or it is unnecessary since we already own the reference?

Reply:
I think __bridge is correct here - we want a CoreFoundation view of `query`, without ownership transfer. If we did __bridge_retained, then we'd have to release the CoreFoundation object too.

At least that's what I understood. :)

Reply:
I meant the `privateKey` cast.

Reply:
I see. CFTypeRef and SecKeyRef are both CoreFoundation, thus outside of ARC management.